File: app/src/main/java/com/spacex/launches/spacexrocketlaunches/about/AboutActivity.java
package com.spacex.launches.spacexrocketlaunches.about;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.NavUtils;
import androidx.databinding.DataBindingUtil;
import com.spacex.launches.spacexrocketlaunches.R;
import com.spacex.launches.spacexrocketlaunches.databinding.AboutBinding;

public class AboutActivity extends AppCompatActivity implements AboutContract.View {

  private AboutBinding viewBinding;
  private AboutContract.Presenter presenter;

  public static void startActivity(@NonNull final Context context) {
    context.startActivity(new Intent(context, AboutActivity.class));
  }

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    viewBinding = DataBindingUtil.setContentView(this, R.layout.about);
    presenter = new AboutPresenter(this);
    setUpToolbar();
  }

  private void setUpToolbar() {
    setSupportActionBar(viewBinding.includeToolbar.toolbar);
    if (getSupportActionBar() != null) {
      getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    }
  }

  @Override
  public boolean onOptionsItemSelected(MenuItem item) {
    switch (item.getItemId()) {
      case android.R.id.home:
        presenter.onUpNavigation();
        return true;
    }
    return super.onOptionsItemSelected(item);
  }

  @Override
  public void navigateUp() {
    NavUtils.navigateUpFromSameTask(this);
  }
}

File: app/src/main/kotlin/com/spacex/ui/about/AboutActivity.kt
package com.spacex.ui.about

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import android.view.View
import androidx.core.app.NavUtils
import androidx.databinding.DataBindingUtil
import com.spacex.ui.R
import com.spacex.ui.common.IntentUtils
import com.spacex.ui.common.isAtLeastLollipop
import com.spacex.ui.databinding.AboutBinding
import com.spacex.ui.di.DaggerAppCompatActivityBase
import javax.inject.Inject

class AboutActivity : DaggerAppCompatActivityBase(), AboutContract.View {

    private lateinit var viewBinding: AboutBinding
    @Inject
    lateinit var presenter: AboutContract.Presenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewBinding = DataBindingUtil.setContentView(this, R.layout.about)
    }

    override fun setUpViews() {
        setUpToolbar()
        viewBinding.clickHandler = AboutActivityClickHandler()
        if (isAtLeastLollipop()) {
            val elevation = resources.getDimension(R.dimen.appbar_elevation)
            viewBinding.header.elevation = elevation
            viewBinding.image.elevation = elevation
        }
    }

    private fun setUpToolbar() {
        setSupportActionBar(viewBinding.includeToolbar.toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                presenter.onUpNavigation()
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }

    override fun navigateUp() {
        NavUtils.navigateUpFromSameTask(this)
    }

    override fun navigateTo(url: String) {
        IntentUtils.loadChromeCustomTab(this, url)
    }

    @Suppress("UNUSED_PARAMETER")
    inner class AboutActivityClickHandler {

        fun onGithubClick(view: View) {
            presenter.onGithubClick()
        }

        fun onStackOverflowClick(view: View) {
            presenter.onStackOverflowClick()
        }

        fun onMediumClick(view: View) {
            presenter.onMediumClick()
        }

        fun onTwitterClick(view: View) {
            presenter.onTwitterClick()
        }

        fun onLinkedInClick(view: View) {
            presenter.onLinkedInClick()
        }
    }

    companion object {

        fun startActivity(context: Context) {
            context.startActivity(Intent(context, AboutActivity::class.java))
        }
    }
}
